core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java

```diff
@@ -76,16 +91,36 @@ public void insertRole(RoleEntity roleEntity, boolean overwritten) {
           MetalakeMetaService.getInstance().getMetalakeIdByName(roleEntity.namespace().level(0));
       RolePO.Builder builder = RolePO.builder().withMetalakeId(metalakeId);
       RolePO rolePO = POConverters.initializeRolePOWithVersion(roleEntity, builder);
+      List<SecurableObjectPO> securableObjectPOS = Lists.newArrayList();
+      for (SecurableObject object : roleEntity.securableObjects()) {
+        SecurableObjectPO.Builder objectBuilder =
+            POConverters.initializeSecurablePOBuilderWithVersion(
+                roleEntity.id(), object, getPOType(object));
+        objectBuilder.withEntityId(
+            getSecurableObjectEntityId(metalakeId, object.fullName(), object.type()));
+        securableObjectPOS.add(objectBuilder.build());
+      }
 
-      SessionUtils.doWithCommit(
-          RoleMetaMapper.class,
-          mapper -> {
-            if (overwritten) {
-              mapper.insertRoleMetaOnDuplicateKeyUpdate(rolePO);
-            } else {
-              mapper.insertRoleMeta(rolePO);
-            }
-          });
+      SessionUtils.doMultipleWithCommit(
+          () ->
+              SessionUtils.doWithoutCommit(
+                  SecurableObjectMapper.class,
+                  mapper -> {
+                    if (overwritten) {
+                      mapper.softDeleteSecurableObjectsByRoleId(rolePO.getRoleId());
+                    }
+                    mapper.batchInsertSecurableObjects(securableObjectPOS);
```

Comment:
need to check `securableObjectPOs` is not empty

Reply:
Done.